SocialCareCaseViewerApi.Tests/V1/IntegrationTests/UpdateAllocationDetails.cs

```diff
@@ -0,0 +1,138 @@
+            updatedAllocation.CreatedBy.Should().Equals(createdAllocation.CreatedBy);
+        }
+
+        [Test]
+        public async Task UpdateAllocationWithDeallocationDetailsUpdatesTheAllocation()
+        {
+            var today = DateTime.Today;
+
+            // Create allocation via API
+            var yesterday = today.AddDays(-1);
+            var patchUri = new Uri("/api/v1/allocations", UriKind.Relative);
+            var newAllocationRequest = new CreateAllocationRequest()
+            {
+                MosaicId = _resident.Id,
+                AllocatedTeamId = _existingDbTeam.Id,
+                AllocationStartDate = yesterday,
+                Summary = "summary",
+                RagRating = "high",
+                CreatedBy = _allocationWorker.Email,
+                CarePackage = "package"
+            };
+            var serializedRequest = JsonSerializer.Serialize(newAllocationRequest);
+            var requestContent = new StringContent(serializedRequest, Encoding.UTF8, "application/json");
+            var createAllocationResponse = await Client.PostAsync(patchUri, requestContent).ConfigureAwait(true);
+
+            // Check the API response and DB records
+            var createdAllocation = DatabaseContext.Allocations.First();
+            createdAllocation.PersonId.Should().Equals(newAllocationRequest.MosaicId);
+            createAllocationResponse.StatusCode.Should().Be(201);
+
+            // Create deallocation request and update created allocation via API
+            var newAllocationPatchRequest = IntegrationTestHelpers.PatchAllocationRequest(
+                allocationId: createdAllocation.Id,
+                createdByWorker: _existingDbWorker,
+                ragRating: null,
+                deallocationReason: "reason",
+                deallocationDate: today);
+            var serializedPatchRequest = JsonSerializer.Serialize(newAllocationPatchRequest);
+            var patchRequestContent = new StringContent(serializedPatchRequest, Encoding.UTF8, "application/json");
+            var patchAllocationResponse = await Client.PatchAsync(patchUri, patchRequestContent).ConfigureAwait(true);
+
+            // Check the API response and DB records
+            var updatedAllocation = DatabaseContext.Allocations.First();
+            patchAllocationResponse.StatusCode.Should().Be(200);
+            updatedAllocation.RagRating.Should().Equals(newAllocationPatchRequest.RagRating);
+            updatedAllocation.Summary.Should().Equals(createdAllocation.Summary);
+            updatedAllocation.CarePackage.Should().Equals(createdAllocation.CarePackage);
+            updatedAllocation.PersonId.Should().Equals(createdAllocation.PersonId);
+            updatedAllocation.AllocationStartDate.Should().Equals(createdAllocation.AllocationStartDate);
+            updatedAllocation.CreatedBy.Should().Equals(createdAllocation.CreatedBy);
+            updatedAllocation.CaseStatus.Should().Equals("CLOSED");
```

Comment:
Not specific to this test, but would be good to use standard casing accross the api based on the current values in prod db i.e. "Open" or "Closed". That'll keep the data consistent